### PR TITLE
Merge dev into master

### PR DESCRIPTION
**Pull request by Grazfather**
* Adds circle config
* Redoes a lot of tests
* Cleans up arches dict
* Does some pylint
* A ton of style cleanup
* Fix a few parsing bugs (Thanks @Tiecoon)
* Add RISCV (Thanks @dlrobertson)
* Improve pie breakpoint parsing (Thanks @d4em0n)

### CODE REVIEW

**Comment by hugsy** [comment]
Wow ! Big big stuff ! I'll review this WE.

File: gef.py [hunk 1]
```diff
@@ -183,8 +183,8 @@ def update_gef(argv):
 except ImportError:
     # if out of gdb, the only action allowed is to update gef.py
     if len(sys.argv)==2 and sys.argv[1]=="--update":
-        sys.exit( update_gef(sys.argv) )
-    gef_print("[-] gef cannot run as standalone")
+        sys.exit(update_gef(sys.argv))
+    print("[-] gef cannot run as standalone")
```
Comment (hugsy) [inline gef.py]: :+1:

File: gef.py [hunk 2]
```diff
@@ -538,6 +538,7 @@ class Elf:
     SPARC             = 0x02
     SPARC64           = 0x2b
     AARCH64           = 0xb7
+    RISCV             = 0xf3
```
Comment (hugsy) [inline gef.py]: How much have you tested RISCV? Do you have a VM ready to use to test it?

Comment (dlrobertson) [reply gef.py]: > How much have you tested RISCV?

Used a few simple examples (similar to the example shown at [Debugging with GDB]) to
verify `is_branch_taken` returned the correct result etc.

> Do you have a VM ready to use to test it?

[riscv-tools] bundles things up nicely, and includes [`riscv-isa-sim`] (`spike`) which enables emulating RISCV.

[Debugging with GDB]: https://github.com/riscv/riscv-isa-sim#debugging-with-gdb
[riscv-tools]: https://github.com/riscv/riscv-tools#riscv-tools-
[`riscv-isa-sim`]: https://github.com/riscv/riscv-isa-sim

Comment (Grazfather) [reply gef.py]: @hugsy I tested this out.See #310.

Comment (hugsy) [reply gef.py]: Then it's a go for me.

**Comment by Grazfather** [comment]
No, was out of town.

**Pull request merged**